pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -573,11 +573,22 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 	// Read the condition the way it was set by the Mark* helpers
 	after = pr.Status.GetCondition(apis.ConditionSucceeded)
 	pr.Status.StartTime = pipelineRunFacts.State.AdjustStartTime(pr.Status.StartTime)
-	pr.Status.TaskRuns = pipelineRunFacts.State.GetTaskRunsStatus(pr)
-	pr.Status.Runs = pipelineRunFacts.State.GetRunsStatus(pr)
+	taskRunStatuses := pipelineRunFacts.State.GetTaskRunsStatus(pr)
+	runStatuses := pipelineRunFacts.State.GetRunsStatus(pr)
+
+	if cfg.FeatureFlags.EmbeddedStatus == config.FullEmbeddedStatus || cfg.FeatureFlags.EmbeddedStatus == config.BothEmbeddedStatus {
+		pr.Status.TaskRuns = taskRunStatuses
+		pr.Status.Runs = runStatuses
+	}
+
+	if cfg.FeatureFlags.EmbeddedStatus == config.MinimalEmbeddedStatus || cfg.FeatureFlags.EmbeddedStatus == config.BothEmbeddedStatus {
+		pr.Status.ChildReferences = pipelineRunFacts.State.GetChildReferences(v1beta1.SchemeGroupVersion.String(),
+			v1alpha1.SchemeGroupVersion.String())
+	}
+
 	pr.Status.SkippedTasks = pipelineRunFacts.GetSkippedTasks()
 	if after.Status == corev1.ConditionTrue {
-		pr.Status.PipelineResults = resources.ApplyTaskResultsToPipelineResults(pipelineSpec.Results, pr.Status.TaskRuns, pr.Status.Runs)
+		pr.Status.PipelineResults = resources.ApplyTaskResultsToPipelineResults(pipelineSpec.Results, taskRunStatuses, runStatuses)
```

Comment:
this looks like a bug: if embedded status is minimal, taskRunStatuses and runStatuses here will be empty slices

Reply:
this doesn't feel right to me. if embedded status is minimal, we're using one function to get the TaskRuns to put into the PipelineRun's status, and a different function to get the TaskRuns to use for TaskRun results?

Reply:
Ok, lemme know if https://github.com/tektoncd/pipeline/pull/4739/commits/e886963dfa53d19c9cd61c86b8810f6e93050aa7 feels good?